[E2E Test] 'go install' requires a version

`run-e2e.sh` installs ginkgo first, but use kusion module:

```bash
# Install ginkgo
GO111MODULE=on go install github.com/onsi/ginkgo/v2/ginkgo
```

However, kusion only depends on `github.com/onsi/ginkgo/v2`, which not include ginkgo's dependencies.
So any contributor run `go  mod tidy` will remove unused entry from go.sum, then pipeline failed:

```bash
# Run e2e test
hack/run-e2e.sh
go: downloading github.com/onsi/ginkgo/v2 v2.0.0
go: downloading github.com/onsi/ginkgo v1.16.5
Error: ../../../go/pkg/mod/github.com/onsi/ginkgo/v2@v2.0.0/ginkgo/generators/bootstrap_command.go:9:2: missing go.sum entry for module providing package github.com/go-task/slim-sprig (imported by github.com/onsi/ginkgo/v2/ginkgo/generators); to add:
	go get github.com/onsi/ginkgo/v2/ginkgo/generators@v2.0.0
Error: ../../../go/pkg/mod/github.com/onsi/ginkgo/v2@v2.0.0/ginkgo/internal/profiles_and_reports.go:12:2: missing go.sum entry for module providing package github.com/google/pprof/profile (imported by github.com/onsi/ginkgo/v2/ginkgo/internal); to add:
	go get github.com/onsi/ginkgo/v2/ginkgo/internal@v2.0.0
Error: ../../../go/pkg/mod/github.com/onsi/ginkgo/v2@v2.0.0/ginkgo/labels/labels_command.go:15:2: missing go.sum entry for module providing package golang.org/x/tools/go/ast/inspector (imported by github.com/onsi/ginkgo/v2/ginkgo/labels); to add:
	go get github.com/onsi/ginkgo/v2/ginkgo/labels@v2.0.0
make: *** [Makefile:35[7](https://github.com/KusionStack/kusion/actions/runs/3410235948/jobs/5672947142#step:7:8): e2e-test] Error 1
Error: Process completed with exit code 2.
```

The true usage as follows:
```bash
$ go install github.com/onsi/ginkgo/v2/ginkgo@latest
go: downloading github.com/onsi/ginkgo/v2 v2.4.0
go: downloading golang.org/x/tools v0.2.0
```

--- a/hack/run-e2e.sh
+++ b/hack/run-e2e.sh
@@ -8,7 +8,7 @@ set -o nounset
 set -o pipefail
 
 # Install ginkgo
-GO111MODULE=on go install github.com/onsi/ginkgo/v2/ginkgo
+GO111MODULE=on go install github.com/onsi/ginkgo/v2/ginkgo@v2.0.0
 
 # Build kusion binary
 go build -o bin/kusion ./cmd/kusionctl/kusionctl.go

